src/controllers/rpc/publish-rpc-controller.js

```diff
@@ -27,21 +27,28 @@ class PublishController extends BaseController {
 
                 break;
             case NETWORK_MESSAGE_TYPES.REQUESTS.PROTOCOL_REQUEST:
-                // eslint-disable-next-line no-case-declarations
-                dataSource = await this.operationIdService.getCachedOperationIdData(operationId);
-                await this.operationIdService.cacheOperationIdData(operationId, {
-                    assertionId: dataSource.assertionId,
-                    assertion: message.data.assertion,
-                });
-                command.name = handleRequestCommand;
-                command.data.keyword = message.data.keyword;
-                command.data.agreementId = dataSource.agreementId;
-                command.data.agreementData = dataSource.agreementData;
+                try {
+                    // eslint-disable-next-line no-case-declarations
+                    dataSource = await this.operationIdService.getCachedOperationIdData(
+                        operationId,
+                    );
+                    await this.operationIdService.cacheOperationIdData(operationId, {
+                        assertionId: dataSource.assertionId,
+                        assertion: message.data.assertion,
+                    });
+                    command.name = handleRequestCommand;
+                    command.data.keyword = message.data.keyword;
+                    command.data.agreementId = dataSource.agreementId;
+                    command.data.agreementData = dataSource.agreementData;
+                } catch (error) {
```

Comment:
When logging an error add more context, where error happened, don't just log error.message.
When publish or any other operation is starting we are emitting start events, if for any reason operation cannot be completed we need to publish end event with error message. Update your code to better handle error in operation